### PR TITLE
Provid totals for 'electricity' entity depending on the type of frequency. 

### PR DESCRIPTION
**Pull request by shahardo**
#291

### CODE REVIEW

**Comment by shahardo** [comment]
Version with working hour chart:
screenshot from 2015-01-29 17 38 23

File: client/app/scripts/services/chartUsage.js [hunk 1]
```diff
@@ -11,36 +62,42 @@ angular.module('negawattClientApp')
...
       var deferred = $q.defer();
 
-      // Limit date to 2 year back
-      var twoYearsBack = Math.floor(Date.now() / 1000) - 2 * 365 * 24 * 60 * 60;
+      // Calculate the time-frame for data request.
+      var chart_frequency = this.usageChartParams.frequency;
+      var chart_frequency_info = this.frequencyParams[chart_frequency];
+      var chart_timeframe = chart_frequency_info.chart_default_time_frame;
+      // @fixme: hard coded timestamp here.
+      var chart_end_timestamp = 1388620800; // Hard code 2/1/2014 for now. Should be: Math.floor(Date.now() / 1000);
```
Comment (amitaibu) [inline client/app/scripts/services/chartUsage.js]: Move to ``this.frequencyParams``

File: client/app/scripts/services/chartUsage.js [hunk 2]
```diff
@@ -11,36 +62,42 @@ angular.module('negawattClientApp')
...
      * @returns {$q.promise}
      */
-    this.get = function(type, id) {
+    this.get = function(selector_type, id) {
       var deferred = $q.defer();
 
-      // Limit date to 2 year back
-      var twoYearsBack = Math.floor(Date.now() / 1000) - 2 * 365 * 24 * 60 * 60;
+      // Calculate the time-frame for data request.
+      var chart_frequency = this.usageChartParams.frequency;
```
Comment (amitaibu) [inline client/app/scripts/services/chartUsage.js]: ``chartFrequency``

File: client/app/scripts/services/chartUsage.js [hunk 3]
```diff
@@ -101,14 +158,20 @@ angular.module('negawattClientApp')
...
+     *    Source object in electricity format.
+     * @param {Object} chart_frequency_info
+     *    Chart frequency info, as defined in this.frequencyParams.
+     *
      * @returns {Object}
-     *   target data in charts' datasets format.
+     *    Target data in google charts' datasets format.
      **/
-    this.transformDataToDatasets = function(data) {
+    this.transformDataToDatasets = function(data, chart_frequency_info) {
```
Comment (amitaibu) [inline client/app/scripts/services/chartUsage.js]: camelCase

File: negawatt/modules/custom/negawatt_restful/plugins/formatter/electricity_hal_json/NegawattFormatterElectricityHalJson.class.php [hunk 4]
```diff
@@ -20,21 +20,6 @@ public function prepare(array $data) {
     // Here we get the data after calling the backend storage for the resources.
     $output = parent::prepare($data);
 
-    // Calculate monthly totals for each meter.
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/formatter/electricity_hal_json/NegawattFormatterElectricityHalJson.class.php]: remove file

**Comment by shahardo** [comment]
Closed in favour of pull request #333

**Pull request closed without merging**